src/plugins/blocksmith/components/blocksmith/error-boundary/index.jsx

```diff
@@ -1,170 +1,99 @@
-          restore editor (including invalid blocks)
-        </Button>
-        )}
-        <p>If you continue to see this message, then we recommend that you restore the valid blocks, save, and continue working in a new tab. Please do not close this tab - support will want to examine it to investigate the error.</p>
-        {!isValid && (
-          <p>The invalid blocks are styled in <span className="invalid-block-indicator"><strong>red</strong></span> below.</p>
-        )}
-        <hr />
-        {blocks.map(({ block }) => {
-          if (block instanceof ContentBlock) {
-            let message = 'block';
-            const blockData = block.getData();
-            if (blockData && blockData.has('canvasBlock')) {
-              message = blockData.get('canvasBlock').schema().getCurie().getMessage();
-            }
-            return (
-              <div className="preview-component preview-component__error" key={block.getKey()}>
-                <p><strong>{`invalid ${message}`}</strong></p>
-                <p>{block.toString()}</p>
-              </div>
-            );
-          }
-          const message = block.schema().getCurie().getMessage();
-          let Component;
-          switch (message) {
-            case 'text-block':
-              // eslint-disable-next-line react/no-danger
-              Component = () => <div dangerouslySetInnerHTML={{ __html: block.get('text') }} />;
-              break;
-            case 'image-block':
-              Component = () => (
-                <>
-                  <p>{message}</p>
-                  <img src={damUrl(block.get('node_ref'), 'o', 'xs')} alt="thumbnail" />
-                </>
-              );
-              break;
-            case 'document-block':
-              Component = () => (
-                <>
-                  <p>{message}</p>
-                  <img src={damUrl(block.get('image_ref'), 'o', 'xs')} alt="thumbnail" />
-                </>
-              );
-              break;
-            case 'gallery-block':
-              Component = () => (
-                <>
-                  <p>{message}</p>
-                  <img src={damUrl(block.get('poster_image_ref', getNode(block.get('node_ref')).get('image_ref')), 'o', 'xs')} alt="thumbnail" />
-                </>
-              );
-              break;
-            case 'video-block':
-              Component = () => (
-                <>
-                  <p>{message}</p>
-                  <img src={damUrl(block.get('poster_image_ref', getNode(block.get('node_ref')).get('image_ref')), 'o', 'xs')} alt="thumbnail" />
-                </>
-              );
-              break;
-            default:
-              Component = () => <p>{message}</p>;
-          }
-          return (
-            <div className="preview-component" key={block.get('etag', block.generateEtag())}>
-              <Component />
-            </div>
-          );
-        })}
-      </div>
+      <Fallback
```

Comment:
@hunyoboy does `Fallback` really need to be a separate component? why not just inline it in index?

Reply:
@efjacobson good morning. it doesn't need to be but i strongly believe that it's beneficial for `ErrorBoundary` to be simple and easy to reason about as much as possible due to the reason that it is our error catcher/protector. any error that'll occur here will bubble up and can cause some issues. plus it makes unit test clearer for assertion. like oh yeah it should render a fallback if error occurs.

Reply:
@hunyoboy it is already simple https://github.com/triniti/cms-js/blob/master/src/plugins/blocksmith/components/blocksmith/error-boundary/index.jsx#L62

having an extra component that is not reused, does not have its own selector/delegate/styles etc is just extra props drilling